astropy/io/fits/util.py

```diff
@@ -824,12 +824,24 @@ def _array_to_file_like(arr, fileobj):
     `numpy.ndarray.tofile`).
     """
 
+    if len(arr) == 0:
```

Comment:
For scientific code especially I prefer `len(...) == 0` because the expectation is more explicit.